lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -172,37 +173,59 @@ def visit_kw(node)
       def visit_params(node)
         node.keywords.each do |keyword,|
           location = keyword.location
-          add_token(location_without_colon(location), :variable)
+          add_token(location_without_colon(location), :parameter)
         end
 
         node.requireds.each do |required|
-          add_token(required.location, :variable)
+          add_token(required.location, :parameter)
         end
 
         rest = node.keyword_rest
-        return if rest.nil? || rest.is_a?(SyntaxTree::ArgsForward)
+        unless rest.nil? || rest.is_a?(SyntaxTree::ArgsForward)
+          name = rest.name
+          add_token(name.location, :parameter) if name
+        end
 
-        name = rest.name
-        add_token(name.location, :variable) if name
+        super
       end
 
       sig { override.params(node: SyntaxTree::VarField).void }
       def visit_var_field(node)
-        case node.value
+        value = node.value
+
+        case value
         when SyntaxTree::Ident
-          add_token(node.value.location, :variable)
+          local = current_environment.find_local(value.value)
+
+          type = if local.nil? || local.type == :variable
+            :variable
+          else
+            :parameter
+          end
```

Comment:
Nitpick:
```suggestion
          type = :variable 
          type = :parameter if local && local.type != :variable
```

Reply:
Actually, it might even be better to extract a method:
```ruby
def type_for_local(value)
  local = current_environment.find_local(value.value)

  if local && local.type != :variable
    :parameter
  else
    :variable
  end
end
```